inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java

```diff
@@ -0,0 +1,388 @@
+import org.apache.inlong.sdk.sort.entity.InLongMessage;
+import org.apache.inlong.sdk.sort.entity.InLongTopic;
+import org.apache.inlong.sdk.sort.entity.MessageRecord;
+import org.apache.inlong.sdk.sort.fetcher.pulsar.PulsarMultiTopicsFetcher;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.RangeAssignor;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Kafka multi topics fetcher
+ */
+public class KafkaMultiTopicsFetcher extends MultiTopicsFetcher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarMultiTopicsFetcher.class);
+    private final ConcurrentHashMap<TopicPartition, OffsetAndMetadata> commitOffsetMap;
+    private final ConcurrentHashMap<TopicPartition, ConcurrentSkipListMap<Long, Boolean>> ackOffsetMap;
+    private final String bootstrapServers;
+    private ConsumerRebalanceListener listener;
+    private KafkaConsumer<byte[], byte[]> consumer;
+
+    public KafkaMultiTopicsFetcher(
+            List<InLongTopic> topics,
+            ClientContext context,
+            Interceptor interceptor,
+            Deserializer deserializer,
+            String bootstrapServers) {
+        super(topics, context, interceptor, deserializer);
+        this.bootstrapServers = bootstrapServers;
+        this.commitOffsetMap = new ConcurrentHashMap<>();
+        this.ackOffsetMap = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public boolean init() {
+        try {
+            this.consumer = createKafkaConsumer();
+            InLongTopic topic = onlineTopics.values().stream().findFirst().get();
+            this.seeker = SeekerFactory.createKafkaSeeker(consumer, topic);
+            this.listener = new AckOffsetOnRebalance(topic.getInLongCluster().getClusterId(), seeker,
+                    commitOffsetMap);
+            consumer.subscribe(onlineTopics.keySet(), listener);
+            return true;
+        } catch (Throwable t) {
+            LOGGER.error("failed to init kafka consumer: {}", t.getMessage(), t);
```

Comment:
Suggested changing to `LOGGER.error("failed to init kafka consumer: ", t);`.

Reply:
fixed, thx